scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/language/json/JsonTemplateParser.scala

```diff
@@ -0,0 +1,181 @@
+    typedExpression.expression match {
+      case expression: SpelExpression =>
+        joinLiteralsAndReplaceExpressionWithDefaults(
+          spelExpressionToDefaultJsonValue(ctx),
+          expression.parsedSpringExpression,
+        )
+      case _ => invalidNel(ExpressionCompilationError("Invalid compiled expression type"))
+    }
+
+  private def joinLiteralsAndReplaceExpressionWithDefaults(
+      spelExpressionToDefaultJsonValue: (String, typing.TypingResult) => ValidatedNel[ExpressionParseError, String],
+      expression: SpringExpression,
+  ): ValidatedNel[ExpressionParseError, String] = expression match {
+    case expression: CompositeStringExpression =>
+      expression.getExpressions.toList
+        .map(e => joinLiteralsAndReplaceExpressionWithDefaults(spelExpressionToDefaultJsonValue, e))
+        .sequence
+        .map(_.mkString)
+    case expression: LiteralExpression => validNel(expression.getValue)
+    case expression: standard.SpelExpression =>
+      spelExpressionToDefaultJsonValue(expression.getExpressionString, Typed[SpelExpressionRepr])
+    case _ => invalidNel(ExpressionCompilationError("Unknown expression type"))
+  }
+
+  private def spelExpressionToDefaultJsonValue(ctx: ValidationContext)(
+      original: String,
+      expectedType: typing.TypingResult
+  ): ValidatedNel[ExpressionParseError, String] = spelParser
+    .parse(original, ctx, expectedType)
+    .map(_.typingInfo.typingResult)
+    .map(typingResultToDefaultJsonValue)
+
+  private def typingResultToDefaultJsonValue(typingResult: TypingResult): String = typingResult match {
+    case TypedClass(k, _) =>
+      k match {
+        case className if TypeValueDeterminer.isIntegerNumber(className)       => "0"
+        case className if TypeValueDeterminer.isFloatingPointNumber(className) => "0.5"
+        case className if TypeValueDeterminer.isBoolean(className)             => "true"
+        case className if TypeValueDeterminer.isString(className)              => "unquoted string"
+        // For now, complex types are treated as String
+        case _ => "unquoted string"
+      }
+    // For now, complex types are treated as String
+    case _ => "unquoted string"
+  }
+
+  private def extractJsonStringFromCompiledExpression(
+      compiledExpression: CompiledExpression
+  ): ValidatedNel[ExpressionParseError, String] =
+    compiledExpression match {
+      case expression: SpelExpression =>
+        joinLiteralsAndReplaceExpressionWithDefaults(
+          spelExpressionToDefaultJsonValue,
+          expression.parsedSpringExpression,
+        )
+      case _ => invalidNel(ExpressionCompilationError("Invalid compiled expression type"))
+    }
+
+  private def spelExpressionToDefaultJsonValue(
+      original: String,
+      expectedType: typing.TypingResult
+  ): ValidatedNel[ExpressionParseError, String] = spelParser
+    .parseWithoutContextValidation(original, expectedType)
+    .map(_ => "0")
```

Comment:
Magic number. Why "0"? I don't understand it.

Reply:
Done - it's because integer works well with and without quotes well